docs/metrics/bleu.md

```diff
@@ -0,0 +1,123 @@
+# BLEU (**B**i**L**ingual **E**valuation **U**nderstudy) Score
+
+The BLEU score is a metric commonly used in a variety of NLP tasks ranging from Machine Translation to Text Summarization, to evaluate the quality of machine-generated text. It quantifies the similarity between the generated text and ground truth text as a score between 0 and 1 — 0 meaning there is no overlap with the ground truth text, and 1 meaning there is a perfect overlap. As the name suggests, it was originally developed for evaluating machine-translation models, but has since been adapted to many different tasks within NLP due to its dynamic nature for measuring textual similarity.
```

Comment:
Similar to the [BERT pr](https://github.com/kolenaIO/kolena/pull/238), I would want the docs to focus on raw text similarity as part of the definition instead of generated text.